Sources/FrontEnd/Parse/Lexer.swift

```diff
@@ -159,10 +165,25 @@ public struct Lexer: IteratorProtocol, Sequence {
       return token
     }
 
-    // Scan numeric literals.
-    if let k = scanNumericLiteral() {
+    // Scan integral literals.
+    if let k = scanIntegralLiteral() {
       token.kind = k
       token.site.extend(upTo: index)
+
+      if let nextChar = peek(), nextChar == "e" || nextChar == "E" {
```

Comment:
```suggestion
      if let next = peek(), next == "e" || next == "E" {
```
There's no need to encode the type of the variable in its name.